extensions/HyperVExtension/src/HyperVExtension/CommunicationWithGuest/GuestKvpChannel.cs

```diff
@@ -237,17 +237,17 @@ private List<IResponseMessage> TryReadResponseMessages(string communicationId, b
 
     private Dictionary<string, string> ReadGuestKvps()
     {
-        return MessageHelper.MergeMessageParts(ReadRawGuestKvps());
+        return MessageHelper.MergeMessageParts(ReadGuestExchangeItems());
     }
 
-    private Dictionary<string, string> ReadRawGuestKvps()
+    private Dictionary<string, string> ReadRawGuestKvps(string itemsType)
     {
         Dictionary<string, string> guestKvps = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
 
         using var collection = _vmWmi.GetRelated("Msvm_KvpExchangeComponent");
         foreach (ManagementObject kvpExchangeComponent in collection)
         {
-            foreach (var exchangeDataItem in (string[])kvpExchangeComponent["GuestExchangeItems"])
+            foreach (var exchangeDataItem in (string[])kvpExchangeComponent[itemsType])

```

Comment:
Do you think it would ever be likely that the `kvpExchangeComponent` doesn't contain the `itemsType` causing it to throw? 

I guess answering my own question, if it does throw that's still fine as we wouldn't want to continue the configuration if we can't figure out the OS details of the VM correct?

Reply:
Yes, if it will throw that would mean we can't continue. Most likely KVP service is not supported or in a non-functional state.